lib/rucio/core/heartbeat.py

```diff
@@ -223,6 +230,57 @@ def list_payload_counts(executable, older_than=600, hash_executable=None, sessio
     return dict(query.all())
 
 
+@read_session
+def get_payload_partition(executable, hostname, pid, thread, payload, older_than=600, hash_executable=None, session=None):
```

Comment:
I don't think this is needed; I would propose to do the `list_and_mark_unlocked_replicas` completely without heartbeats, as it does not have any added benefit. And in that case, we do not need to do payload_partitions.